core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveDeltaFloatingCodec.java

```diff
@@ -225,6 +238,71 @@ public double decodeDouble(long value) {
       return (max - value) / factor;
     }
 
+    @Override
+    public void decodeAndFillVector(ColumnPage columnPage, ColumnVectorInfo vectorInfo) {
+      CarbonColumnVector vector = vectorInfo.vector;
+      BitSet nullBits = columnPage.getNullBits();
+      DataType type = columnPage.getDataType();
+      int pageSize = columnPage.getPageSize();
+      BitSet deletedRows = vectorInfo.deletedRows;
+      DataType dataType = vector.getType();
```

Comment:
What's the relationship between `columnPage.getDataType()` and `vector.getType()`?

Reply:
column page datatype is adpative datatype and vector datatype is actual type. I will rename them